server/opts.go

```diff
@@ -575,17 +593,73 @@ func parseSubjects(v interface{}) ([]string, error) {
 	default:
 		return nil, fmt.Errorf("Expected subject permissions to be a subject, or array of subjects, got %T", v)
 	}
-	return checkSubjectArray(subjects)
```

Comment:
Why we don't check for validity here?

Reply:
You are correct, need to rework this and use parseSubject() more. I see duplicate code. One sec.